pkg/certificate/mrc_reconciler.go

```diff
@@ -170,10 +181,36 @@ func (m *Manager) updateMRCState(mrc *v1alpha2.MeshRootCertificate) error {
 		mrc.Status.TransitionAfter = nil
 	} else {
 		mrc.Status.TransitionAfter = &metav1.Time{Time: time.Now().Add(mrcDurationPerStage)}
+		log.Debug().Str("mrc", namespacedMRCName(mrc)).Msgf("updated MRC TransitionAfter to %s", mrc.Status.TransitionAfter)
+	}
+
+	if m.shouldRetrieveCA(mrc) {
```

Comment:
nit: This name was a little misleading, it is more "ensureIssuerforMRC" I think?

Reply:
Good point, I agree